### PR TITLE
Fix the link to customers table

### LINKED ISSUE #3037: Customer link on Transactions page gives unexpected results
### Describe the bug
[customer_url](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/wc-payment-api/class-wc-payments-api-client.php#L1493) in transaction page is formed using `$order->get_customer_id()`, which returns 0 for guest customers and the value of `_customer_user` order meta key for logged in users. However, customer ids used in WooCommerce -> Customers table are different, which makes the above link give either incorrect or no results. 

### To Reproduce

1. Make a purchase as a guest
2. Go to WooCommerce Payments -> Transactions page
3. Click on Customer link in the relevant transaction entry
4. Note that the link returns an empty/incorrect result in Customers table

### Actual behavior

Clicking on Customer link on a transaction leads to an empty or incorrect result

### Screenshots

https://user-images.githubusercontent.com/6216000/135883987-f2444367-bf5e-4387-8af3-84687c72a621.mov

### Expected behavior
Clicking on Customer link on a Transaction in Transactions table should show the relevant customer in the WooCommerce Customers table

### Additional context
https://github.com/Automattic/woocommerce-payments/pull/3030#issuecomment-932683842

### PR DESCRIPTION
**Pull request by KarlisJ**
Fixes #3037 

#### Changes proposed in this Pull Request

**WooCommerce -> customers** table reads records from `wp_wc_customers_lookup` table but when building link from `includes/wc-payment-api/class-wc-payments-api-client.php` we are passing id that is read from `wp_users` table. 

We're using `$order->get_customer_id()` method but it returns id from `wp_users` table. If guest checkouts are turned off this causes no visual harm as id's in both tables are coincide but once guess checkout is allowed two potential problems may occur:
- For guess customers given method returns `0`
- For non-guess customers method can return incorrect id if there have been previous guest checkouts

The proposed solution uses an available method from the WC-Admin plugin that takes care of fetching the correct customer id.

During the investigation, we found another issue. With Analytics disabled we don't create entries in `wp_wc_customers_lookup` table but once Analytics is enabled we are trying to link to the non-existing customer. The solution to this problem is simply not displaying the entry as link if we can't find customer id.

#### Testing instructions

1. Make sure that Accounts & Privacy settings for WooCommerce allow for guest checkout
2. Make a purchase as a guest
3. Make a purchase from an account (or create during checkout)
4. Make sure that `wc-admin_import_customers` scheduled action from **WooCommerce -> status** are run
4. Go to **Payments -> transactions*
5. Check that for both transactions the customer link points to the correct customer
6. Go to **WooCommerce -> settings** 'Advanced' tab under 'features' uncheck "Enable Analytics".
7. Repeat steps 2. - 5. 

-------------------

- [x] Added changelog entry (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (does not apply)

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * Fix - Use configured domain instead of current domain for Apple Pay verification.
 * Fix - Fatal error when deactivating the WooCommerce plugin when WCPay Subscriptions is enabled.
 * Fix - Do not create WooCommerce Payments Subscriptions when using payment methods other than WooCommerce Payments.
+* Fix - Use reports customer id for customers table.
```
Comment (anu-rock) [inline changelog.txt:13]: The wordings do not sufficiently explain the changes and may benefit from a little rephrasing.

**Comment by anu-rock** [comment]
Congratulations, @KarlisJ, on your first PR 🎉  
Great job with the research that helped you uncover a single-line code change.

The testing instructions are nicely detailed out. I was able to follow through the two scenarios. I confirm that the fix works!

I've left a minor comment wrt the changelog entry. Besides that, if there's an opportunity to write unit tests then we should add them.

Also looping in @kalessil and @dmvrtx who have more context of Woo core than me to validate there's no major impact.

**Comment by dmvrtx** [comment]
While reviewing this PR I found another issue: if "WooCommerce analytics" is not enabled, there is no "Customers" page route registered with WooCommerce and link from the "Transactions" page lands on an empty content.

**Comment by dmvrtx** [comment]
:x: **Case 1:** Customer checks out without creating an account, WooCommerce analytics disabled.

On Transactions page link leads to `.wp-admin/admin.php?page=wc-admin&path=%2Fcustomers&filter=single_customer` which is eventually an empty page.

With WooCommerce analytics enabled content appears on the page, but instead of a specific customer details there is a list with customers.

:white_check_mark: **Case 2:** Customer checks out without creating an account, WooCommerce analytics enabled.

Customer table contains a new record, link from the Transactions page works as expected.

:white_check_mark: **Case 3:** Customer checks out with an account, WooCommerce analytics enabled.

Customer table contains a new record, link from the Transactions page works as expected.

:x: **Case 4:** Customer checks out with an account, WooCommerce analytics disabled.

Transactions page has a link with customer id, however it brings us to an empty page.

---

I am not sure if it should be part of this PR or a separate issue but we have to make our Transactions page to better handle disabled WooCommerce analytics.

**Review by kalessil — APPROVED** [review]
LGTM !

I did some extra checks:
- The method is available in the supported WC versions
- Binding to WC Admin is acceptable in this case.

**Comment by kalessil** [comment]
Good catch @dmvrtx, we could replicate some logic from the store method: if order retrieves an empty customer id, try looking up it by email (likely there is a WC function/method for this). If customer id is still not identified, render just 'guest' instead of the link.

**Comment by KarlisJ** [comment]
@dmvrtx thank you for catching this. 

I agree with @kalessil about rendering just 'guest' instead of link but there's no need to create any lookup by email as `\Automattic\WooCommerce\Admin\API\Reports\Customers\DataStore::get_existing_customer_id_from_order` already handles that logic. The problem here is that when analytics are disabled we don't create an entry in `wp_wc_customers_lookup` table thus there's nothing to look up and display.

I believe there will be more issues like this because the term "Customer" is being used in two different ways though out the codebase. In most cases it's referring to user account which from backend perspective is entry in `wp_users`. In some cases it refers to Analytics customers. 

Ideally, we should use a term like `analytics_customer` to remove any possible confusion but I'm not sure how realistic it would be to carry out the refactor at this stage.

**Comment by kalessil** [comment]
@KarlisJ: so if customer id can be accessed from the order, we render the link and 'guest' otherwise. Do I get it right?

If I got it right, let's go ahead with this approach - I like the simplicity of it.

**Comment by KarlisJ** [comment]
@kalessil Yes the approach is clean and simple but re-thinking I don't like the 'guest' in this case as it will only increase the confusion for the end-user. 

It all comes down to having two types of customers. End-users think of guest checkout as placing an order without creating an account. But analytics create an entry in `wp_wc_customers_lookup` for every checkout, including guest checkouts.

If the order doesn't have an associated entry in `wp_wc_customers_lookup` it doesn't necessarily mean it's a guest checkout (might have been anything). It means that analytics were disabled (or for any other reason the entry wasn't created) and we have nothing to display. **So short term I think it's best to simply display blank for these cases.** 

We could try to fetch the full name from `wp_users` table and display it as simple non-clickable text but we could do that only for non-guest checkouts. This would result in even bigger confusion once analytics are disabled and guest checkout allowed. 

---
Long term we should resolve the confusion around the term "customer". For example, for guest checkout there's 'Guest' under the 'Customer' field in the order page but a clickable link to customer in the WCpay transactions page (because analytics create customer entry for every checkout). Since both say "Customer" it's confusing for the end-user (and developers).

**Review by anu-rock — APPROVED** [review]
LGTM 🚢

File: client/transactions/list/index.tsx [hunk 2]
```diff
@@ -221,13 +221,14 @@ export const TransactionsList = (
...
-			txn.customer_name
-		);
+		const customerName =
+			txn.order && txn.order.customer_url ? (
+				<Link href={ txn.order.customer_url ?? '' }>
+					{ txn.customer_name }
+				</Link>
+			) : (
+				txn.customer_name
+			);
```
Comment (KarlisJ) [inline client/transactions/list/index.tsx:231]: @kalessil No need to reach for  `get_formatted_billing_full_name` and `get_billing_email` or other methods. The `customer_name` is already available on the frontend.

Comment (kalessil) [reply client/transactions/list/index.tsx:231]: @KarlisJ, awesome!

**Pull request merged**